unime/src/routes/+layout.svelte

```diff
@@ -39,23 +39,24 @@
 
   const systemColorScheme = window.matchMedia('(prefers-color-scheme: dark)');
 
-  systemColorScheme.addEventListener(
-    'change',
-    (e) => {
-      if ($state?.profile_settings.profile?.theme) {
-        determineTheme(e.matches, $state.profile_settings.profile.theme);
-      }
-    },
-    { once: true },
-  );
+  // This is used for detecting the OS theme changes.
+  systemColorScheme.addEventListener('change', (e) => {
+    if ($state?.profile_settings.profile?.theme) {
+      determineTheme(e.matches, $state.profile_settings.profile.theme);
+    } else {
+      determineTheme(systemColorScheme.matches, 'system');
+    }
+  });
 
   $: {
     // TODO: needs to be called at least once to trigger subscribers --> better way to do this?
     console.log('+layout.svelte: state', $state);
 
-    // needed again?
+    // This is used for detecting the theme changes inside the app.
```

Comment:
I don't think this comment adds any value since the check is already descriptive: `$state` > `profile_settings` > `theme`.